src/module/orbbec.cpp

```diff
@@ -1175,48 +1186,22 @@ vsdk::Camera::image_collection Orbbec::get_images(std::vector<std::string> filte
                 throw std::invalid_argument(buffer.str());
             }
 
-            if (color->getFormat() != OB_FORMAT_MJPG) {
-                throw std::invalid_argument("color frame was not in jpeg format");
-            }
-
-            std::optional<DeviceFormat> res_format_opt;
-            {
-                std::lock_guard<std::mutex> lock(config_by_serial_mu());
-                if (config_by_serial().count(serial_number) == 0) {
-                    throw std::invalid_argument("device with serial number " + serial_number + " is not in config_by_serial");
-                }
-                res_format_opt = config_by_serial().at(serial_number).device_format;
-            }
-            if (res_format_opt.has_value()) {
-                if (res_format_opt->color_format.has_value()) {
-                    if (ob::TypeHelper::convertOBFormatTypeToString(color->getFormat()) != res_format_opt->color_format.value()) {
+            if (device_format_opt.has_value()) {
+                if (device_format_opt->color_format.has_value()) {
+                    if (ob::TypeHelper::convertOBFormatTypeToString(color->getFormat()) != device_format_opt->color_format.value()) {
                         std::ostringstream buffer;
-                        buffer << "color format mismatch, expected " << res_format_opt->color_format.value() << " got "
+                        buffer << "color format mismatch, expected " << device_format_opt->color_format.value() << " got "
```

Comment:
so this `device_format_opt` is something we get back from the SDK and it can tell us what formats are actually available instead of hardcoding it as we did before? Is this right? Checking my understanding here